articles/openshift/oshift-ref-cluster-monitoring.md

```diff
@@ -0,0 +1,53 @@
+---
+title: OpenShift integrated cluster monitoring | UKCloud Ltd
+description: Provides information on the Prometheus monitoring stack deployed with OpenShift v3.11
+services: openshift
+author: Kieran O'Neill
+toc_rootlink: Reference
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: OpenShift integrated cluster monitoring
+toc_fullpath: Reference/oshift-ref-cluster-monitoring.md
+toc_mdlink: oshift-ref-cluster-monitoring.md
+---
+
+# OpenShift integrated cluster monitoring
+
+### Overview
+
+OpenShift 3.11 ships with a pre-configured and self-updating monitoring stack based on the Prometheus open source project. This ships with a set of alerts that notify the cluster admin of problems and a set of Grafana dashboards. The deployed Prometheus instance is responsible for monitoring and alerting on cluster and OpenShift components, it shouldn't be extended to monitor user applications. Users interested in leveraging Prometheus for application monitoring on OpenShift should consider using the operator lifecycle manager (https://github.com/operator-framework/operator-lifecycle-manager) to easily deploy a Prometheus Operator and setup new Prometheus instances to monitor and alert on their applications. Alertmanager is a cluster-global component for handling alerts generated by all Prometheus instances deployed in that cluster.
+
+### Resource
+
+We run cluster monitoring with persistent storage, this means that the metrics are stored on a persistent volume and can survive a pod being restarted or recreated. Each Alertmanager instance conumes 2Gi and each Prometheus instance consumes 50Gi. In a standard cluster with two infrastructure nodes this will mean 106Gi of storage will be used for cluster monitoring. If the persistent volumes fill up, or you notice them getting close to being full, you can follow this guide to expand the persistent volumes: https://docs.ukcloud.com/articles/openshift/oshift-how-expand-persistent-vols.html
+
+### Configuration
+
+The only Red Hat supported configuration options for the integrated monitoring stack are variables used at the time of deployment and can be seen here: https://docs.openshift.com/container-platform/3.11/install_config/prometheus_cluster_monitoring.html#configuring-openshift-cluster-monitoring
```

Comment:
I don't like this approach. Read the options, most are not customer relevant and would require us to do something like change the storage class tier or something. The exception is the alertmanager config which is probably passed into a configmap or something that is customisable. Investigate that a bit more and we might be able to put some guidance in there. These docs need to add value too not just summarise the RH docs.

Reply:
Why couldn't we redeploy Prometheus with tier1 storage/more storage etc if the customer asks for it? Mostly that was put in to make the following make sense: 'Red Hat advises: Beyond those explicit configuration options, it is possible to inject additional configuration into the stack. However this is unsupported, as configuration paradigms might change across Prometheus releases, and such cases can only be handled gracefully if all configuration possibilities are controlled.' I'll take a look at the alertmanage stuff.